src/qmapshack/gis/fit/decoder/CFitFieldDataState.cpp

```diff
@@ -133,7 +133,20 @@ void CFitFieldDataState::devProfile(CFitMessage& mesg)
     // part 1, dynamic profiles creation
     if (mesg.getGlobalMesgNr() == eMesgNumDeveloperDataId)
     {
-        clearDevFieldProfiles();
+        // Get developer ID
+        quint8 devDataIdx = fitDevDataIndexInvalid;
+        for (const CFitField &field : mesg.getFields())
```

Comment:
I don't get it. There is a const reference!? `const CFitField &field`

Reply:
```
        const QList<CFitField>& fields = mesg.getFields();
        for (const CFitField &field : fields)

```
Else the range loop creates another copy.

Reply:
BTW I never got a clazy-range-loop warning from clazy, even with the highest level actived.

Reply:
Sure that clazy is working properly?

Reply:
At least it will do something. I have 
```
:~/tmp$ clazy --version
clazy version: 1.10
clang version: 10.0.0-4ubuntu1
```
It is compiled by me because the clazy version which comes with the system does not work.

Checking the file you see in the editor I got the following output, activated the highest level of clazy

![image](https://user-images.githubusercontent.com/37560190/114772083-ad55a780-9d6d-11eb-923f-462314115497.png)

Reply:
I would recommend to install qtcerator form https://www.qt.io/ 

The distribution stuff usually is outdated ore has a really weird setup